Neos.Media/Classes/Fusion/ImageImplementation.php

```diff
@@ -11,19 +11,20 @@
  * source code.
  */
 
+use Exception;
 use Neos\Flow\Annotations as Flow;
+use Neos\Fusion\FusionObjects\DataStructureImplementation;
 use Neos\Media\Domain\Model\AssetInterface;
 use Neos\Media\Domain\Model\ThumbnailConfiguration;
 use Neos\Media\Domain\Service\AssetService;
-use Neos\Fusion\FusionObjects\AbstractFusionObject;
 use Neos\Media\Domain\Service\ThumbnailService;
 
 /**
  * Render an AssetInterface: object. Accepts the same parameters as the uri.image ViewHelper of the Neos.Media package:
  * asset, width, maximumWidth, height, maximumHeight, allowCropping, allowUpScaling.
  *
  */
-class ImageUriImplementation extends AbstractFusionObject
+class ImageImplementation extends DataStructureImplementation
```

Comment:
This should still extend the AbstractFusionObject inheriting from DataStructureImplementation does not offer any advantage.

Reply:
I know that is really nitpicky, but fusion is such a public thing, I fear this might be a breaking change for people extending the implementation, and while I am usually also the first to say, so what, I have the feeling that fusion implementations are probably a prime target of being extended, just because you can replace them so easily.

Reply:
So should I leave it with that or switch back to `AbstractFusionObject`?

Reply:
Hmm, not sure 

we might also just add the new Prototypes to Media and simply don’t touch the old ones but deprecate then. For Neos 6 we can then remove the implementations and add a migration to the new names and then in Neos 7 finally remove the Prototypes.

@christian would that be a valid strategy

Reply:
I guess the full blown b/c with migration path strategy would be empty classes extending the existing ones + empty prototypes and deprecating the old classes and prototypes. Then move actual code over in next major and delete old stuff.

Reply:
Then I'd go for that. Means recreating the classes and extend from the new ones. So I guess I will revert the commit where I moved the class and then do that. And add the migration.

Just how can I mark a Fusion prototype deprecated?

Reply:
But when extending the old class from the new one, should I do the following?

```
public function evaluate(): string
{
   return super::evaluate()['src'];
}
```

Because the new class returns the full array whereas the old class just returned the src string.

Reply:
> Just how can I mark a Fusion prototype deprecated?
Add a deprecated notice to he source and the fusion docs

Reply:
I am fine with the inheritence now but it would be good to use AbstractFusionObject as base and not DataStructureImplementation. The DataStructure implements a lot of stuff we do not want / need here.